examples/routing_service/routing_service_file_adapter/cpp/FileConnection.cxx

```diff
@@ -0,0 +1,73 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include "FileConnection.hpp"
+#include "FileStreamReader.hpp"
+#include "FileStreamWriter.hpp"
+
+using namespace rti::community::examples;
+using namespace rti::routing;
+using namespace rti::routing::adapter;
+
+FileConnection::FileConnection(
+        StreamReaderListener *input_stream_discovery_listener,
+        StreamReaderListener *output_stream_discovery_listener,
+        const PropertySet &properties) :
+        input_discovery_reader_(properties, input_stream_discovery_listener)
+{
+};
+
+StreamReader *FileConnection::create_stream_reader(
+        Session *session,
+        const StreamInfo &info,
+        const PropertySet &properties,
+        StreamReaderListener *listener)
+{
+    return new FileStreamReader(this, info, properties, listener);
+};
+
+void FileConnection::delete_stream_reader(StreamReader *reader)
+{
+    delete reader;
+}
+
+StreamWriter *FileConnection::create_stream_writer(
+        Session *session,
+        const StreamInfo &info,
+        const PropertySet &properties)
+{
+    return new FileStreamWriter(properties);
+};
+
+void FileConnection::delete_stream_writer(StreamWriter *writer)
+{
+    delete writer;
+}
+
+DiscoveryStreamReader *FileConnection::output_stream_discovery_reader() 
```

Comment:
I'm not sure if we need to implement this one, since it shouldn't be a pure virtual method. In such case, let's not define it nor declare it

Reply:
We can discuss this tomorrow. I would want to keep it for the purpose of having skeleton code if a customer is writing their own adapter.